packages/compiler/core/checker.ts

```diff
@@ -3219,6 +3252,21 @@ export function createChecker(program: Program): Checker {
     return type;
   }
 
+  function resolveTypeReferenceString(reference: string): Type | undefined {
+    const parser = createParser(reference);
+    const node = parser.parseStandaloneReferenceExpression();
+    if (parser.parseDiagnostics.length > 0) {
```

Comment:
I don't think the parser has warnings yet, but I wonder what is right here if there are warnings. Actually, I'm OK with being conservative. I would want to fix my parse warnings if I called this. But see also throw vs report below.

Reply:
hhm actually it might emit a warning if referencing a deprecated type which then with the `throw` would prevent you from doing that.